### DISCUSSION TITLE
Ready to buy the full version ... Had forever, used seldomly, Just became essential.. but first...

### DISCUSSION THREAD

**Discussion by stujumpsrope** [rxhanson/Charmstone-Community#52] [Q&A]
Hi Ryan
This app sat in my menubar until, just very recently, an 'ah-ha' moment occurred and it became the unmissable Magic Trackpad -for-the One-Handed-MacOS user who is over-caffeinated and slouching back after very long two-handed sessions.  Where I originally had use for only four slots for apps/locations, now I need more!

Could you please answer:

- Is the app still in development?  posts like [here](https://github.com/rxhanson/Charmstone-Community/discussions/32#discussioncomment-6316986)  allude to feature updates.

- Does it have to be sandboxed / only available on the App store?  I have a work computer and personal with 2 iCloud accounts and prefer to purchase once.

- a) These Black Friday sales are starting, do you ever discount your apps?  Will you? Might you consider? 
- b) Seriously, if you offered a bundle discount for all your apps I would have bought already.  I love Rectangle Pro.  It's my number one.  Very "detailed", _user-courteous_, _graceful_ app development - the documentation too.  Other high-output App devs should take note from you.  I would buy all your apps even if I seldom used but that's too much money.  That's why I'm asking, or suggesting and passing on the praise just cos I can.

Thankyou.

**Comment by rxhanson** [comment]
Thanks for the feedback!

> Is the app still in development? posts like https://github.com/rxhanson/Charmstone-Community/discussions/32#discussioncomment-6316986 allude to feature updates.

Currently I'm fixing bugs and am getting ready to push out an update soon. I use Charmstone all the time, and although I have a lot of ideas that I would like to implement, I also really like keeping the app slim and view the simplicity as a feature. I don't view this as "no more features ever", I'm just picky about what I want in this app. Do you have feature ideas that you consider to be a dramatic improvement?

> Does it have to be sandboxed / only available on the App store? I have a work computer and personal with 2 iCloud accounts and prefer to purchase once.

There's a non-App Store version at https://charmstone.app, and you'll want to purchase outside of the App Store if possible because I've found that sometimes the family sharing of in-app-purchases doesn't always work. At this time, the App Store version and the non-App Store version are separate, so a purchase in one would not transfer to a purchase in the other.

> a) These Black Friday sales are starting, do you ever discount your apps? Will you? Might you consider?

I typically never discount any of my apps. I might do this at some point in the future, but I have no plans to discount any of them for Black Friday sales this year (sorry!)

> b) Seriously, if you offered a bundle discount for all your apps I would have bought already. I love Rectangle Pro. It's my number one. Very "detailed", user-courteous, graceful app development - the documentation too. Other high-output App devs should take note from you. I would buy all your apps even if I seldom used but that's too much money. That's why I'm asking, or suggesting and passing on the praise just cos I can.

Thanks for the kind words! There is a bit of a limitation between in/out of the App Store bundling, and some apps kind of don't make sense to bundle together (Multitouch includes the Middle functionality, Scroll solves problems that are only faced by certain users, etc). I agree, though, there's likely some bundles here that make sense to offer.

**Comment by stujumpsrope** [comment]
Thanks for you reply Ryan.

I can think of an awesome app bundle: Rectangle Pro, Charmstone, Hightop, Multi-touch, and of course, HyperKey,  despite other automation apps offering to do it for me. I use or have to the side, all of these. I'm a big fan of your trackpad gestures in rectangle: resting three touches and removing one finger, as it simply works. If I hadn't committed to other trackpad movement apps, I would be using Multi-touch.  

Can I cloud-sync the settings so I can use across computers with different AppleIDs?  I have a work computer and AppleID.  Even if I go through the back door by moving a preferences folder to a cloud sync service and sym-link to it from within Charmstone?

As for feature ideas, yes. Apart from preference syncing, could Charmstone display a folder with only a certain tag?  Use case is my project folder I'm currently working on is tagged with green in Finder, other projects are tagged red.  It would be great to have this dynamically populate the Charmstone circle.  I currently manually add the latest project folder to Charmstone which stays there for the duration of the project.  I'm looking to play around with the scripting-within-an-app feature; is this something it could do?

Thanks once again.

**Comment by rxhanson** [comment]
> I can think of an awesome app bundle: Rectangle Pro, Charmstone, Hightop, Multi-touch, and of course, HyperKey, despite other automation apps offering to do it for me. I use or have to the side, all of these. I'm a big fan of your trackpad gestures in rectangle: resting three touches and removing one finger, as it simply works. If I hadn't committed to other trackpad movement apps, I would be using Multi-touch.

Good to know, thanks!

> Can I cloud-sync the settings so I can use across computers with different AppleIDs? I have a work computer and AppleID. Even if I go through the back door by moving a preferences folder to a cloud sync service and sym-link to it from within Charmstone?

This has been on my todo list for a while. One thing that's held me back from doing it is that the App Store version of Charmstone is sandboxed and only has access to files/folders that you give it. Therefore, the files/folders selections in Charmstone wouldn't be transferred and would have to be manually entered despite all of the app selections being transferred, and I find it annoying to have something that only partially works. Perhaps for the non-AppStore version, I can get away with doing this better, but I prefer to keep parity between the two at this time. It's also oddly difficult to get iCloud sync working really well. I'll add your +1 to this, though.

> As for feature ideas, yes. Apart from preference syncing, could Charmstone display a folder with only a certain tag? Use case is my project folder I'm currently working on is tagged with green in Finder, other projects are tagged red. It would be great to have this dynamically populate the Charmstone circle. I currently manually add the latest project folder to Charmstone which stays there for the duration of the project. I'm looking to play around with the scripting-within-an-app feature; is this something it could do?

This is another sort of issue with only having access to folders that you specify. Technically you might be able to use symlinks to achieve this, but I've never tried that. Maybe there's another way to achieve this?
